src/providers/UploadProvider.js

```diff
@@ -0,0 +1,217 @@
+// @flow
+import { useNavigation } from "@react-navigation/native";
+import type { Node } from "react";
+import React, { useCallback, useMemo, useState } from "react";
+import Observation from "realmModels/Observation";
+import ObservationPhoto from "realmModels/ObservationPhoto";
+import { formatDateAndTime } from "sharedHelpers/dateAndTime";
+import fetchPlaceName from "sharedHelpers/fetchPlaceName";
+
+import { UploadContext } from "./contexts";
+
+type Props = {
+  children: any
+}
+
+const ObsEditProvider = ( { children }: Props ): Node => {
+  const navigation = useNavigation( );
+  const [currentObsIndex, setCurrentObsIndex] = useState( 0 );
+  const [observations, setObservations] = useState( [] );
+  const [cameraPreviewUris, setCameraPreviewUris] = useState( [] );
+  const [galleryUris, setGalleryUris] = useState( [] );
+  const [evidenceToAdd, setEvidenceToAdd] = useState( [] );
+
+  const resetUploadContext = useCallback( ( ) => {
+    setObservations( [] );
+    setCurrentObsIndex( 0 );
+    setCameraPreviewUris( [] );
+    setGalleryUris( [] );
+    setEvidenceToAdd( [] );
+  }, [] );
+
+  const allObsPhotoUris = useMemo(
+    ( ) => [...cameraPreviewUris, ...galleryUris],
+    [cameraPreviewUris, galleryUris]
+  );
+
+  const currentObs = observations[currentObsIndex];
+
+  const addSound = async ( ) => {
+    const newObs = await Observation.createObsWithSounds( );
+    setObservations( [newObs] );
+  };
+
+  const addObservations = async obs => setObservations( obs );
+
+  const createObservationNoEvidence = async ( ) => {
+    const newObs = await Observation.new( );
+    setObservations( [newObs] );
+  };
+
+  const createObsPhotos = useCallback( async photos => Promise.all(
+    photos.map( async photo => ObservationPhoto.new( photo?.image?.uri ) )
+  ), [] );
+
+  const createObservationFromGalleryPhoto = useCallback( async photo => {
+    const latitude = photo?.location?.latitude || null;
+    const longitude = photo?.location?.longitude || null;
+    const placeGuess = await fetchPlaceName( latitude, longitude );
+    // create a new observation using the data in the first grouped photo
+    // TODO: figure out if we want to loop through observations, looking for one
+    // with lat/lng, if the first photo lat/lng is blank
+    const newObs = {
+      latitude,
+      longitude,
+      time_observed_at: formatDateAndTime( photo.timestamp ),
+      place_guess: placeGuess
+    };
+    return Observation.new( newObs );
+  }, [] );
+
+  const createObservationsFromGroupedPhotos = useCallback( async obs => {
```

Comment:
Can we stop using "obs" as an abbreviation? I always get confused about whether it means a single observation or multiple observations.